server/src/internalClusterTest/java/org/opensearch/snapshots/RestoreSnapshotIT.java

```diff
@@ -1055,4 +1058,48 @@ public void testRestoreBalancedReplica() {
         }
     }
 
+    public void testRestoreWithReplicationTypeSettings() throws Exception {
```

Comment:
I think we have this case covered in https://github.com/opensearch-project/OpenSearch/blob/7ffcd6500e0bd5956cef5c289ee66d9f99d533fc/server/src/internalClusterTest/java/org/opensearch/snapshots/SegmentReplicationSnapshotIT.java#L128